tests/eas/misfit.py

```diff
@@ -0,0 +1,400 @@
+            "events" : [
+                "sched_switch",
+                "sched_wakeup",
+                "sched_wakeup_new",
+                "cpu_idle",
+            ],
+            "buffsize" : 100 * 1024
+        },
+        "modules": ["cgroups", "cpufreq"],
+    }
+
+    @classmethod
+    def setUpClass(cls, *args, **kwargs):
+        super(_MisfitMigrationBase, cls).runExperiments(*args, **kwargs)
+
+    @memoized
+    @staticmethod
+    def _get_cpu_capacity(test_env, cpu):
+        """
+        :returns: The capacity value of a given CPU on the target
+        """
+        if test_env.nrg_model:
+            return test_env.nrg_model.get_cpu_capacity(cpu)
+
+        return test_env.target.read_int(
+            '/sys/devices/system/cpu/cpu{}/cpu_capacity'.format(cpu))
+
+    @memoized
+    @staticmethod
+    def _get_sd_info(test_env, cpu):
+        """
+        Read the sysfs entry of a CPU's sched domains
+
+        :returns: a dict of dicts. The first level is the integer representing
+        the sched_domain index (sched_domain<key>/* entries), the second level
+        is the sched domain field (sched_domainX/<key> entry)
+        """
+        data = test_env.target.read_tree_values(
+            '/proc/sys/kernel/sched_domain/cpu{}/'.format(cpu), depth=2
+        )
+
+        return {int(key.lstrip('domain')) : values for key, values in data.iteritems()}
+
+    @memoized
+    @staticmethod
+    def _get_hz(test_env):
+        """
+        :returns: The value of CONFIG_HZ on the target
+        """
+        return int(test_env.target.config.get('CONFIG_HZ'))
+
+    @memoized
+    @classmethod
+    def _has_asym_cpucapacity(cls, test_env):
+        """
+        :returns: Whether SD_ASYM_CPUCAPACITY is set on any sd in the system
+        """
+        for cpu in xrange(test_env.platform['cpus_count']):
+            for domain, data in cls._get_sd_info(test_env, cpu).iteritems():
+                if int(data['flags']) & SD_ASYM_CPUCAPACITY == SD_ASYM_CPUCAPACITY:
+                    return True
+
+        return False
+
+    @memoized
+    @classmethod
+    def _classify_cpus(cls, test_env):
+        """
+        Classify cpus per capacity.
+
+        :returns: A list of list. CPUs of equal capacities are packed in the
+        same list, and those lists of CPUs are ordered by capacity values.
+        """
+        cpus = {}
+        for cpu in xrange(test_env.platform['cpus_count']):
+            cap = cls._get_cpu_capacity(test_env, cpu)
+            if cap not in cpus:
+                cpus[cap] = []
+
+            cpus[cap].append(cpu)
+
+        capacities = sorted(cpus.keys())
+        return [cpus[capacity] for capacity in capacities]
+
+    @memoized
+    @classmethod
+    def _get_max_lb_interval(cls, test_env):
+        """
+        :returns: The absolute maximum load-balance interval in seconds
+        """
+        HZ = cls._get_hz(test_env)
+        return ((HZ * test_env.platform['cpus_count']) / 10) * (1. / HZ)
```

Comment:
Is that hardcoded `/10` intentional?... I guess yess... can we add a bit of doc explaining why 10?

And.. does it makes sense for this also to be a devlib API?

Reply:
Yeah it's defined like that in the kernel:
```c
max_load_balance_interval = HZ/10;
```

Reply:
Then let's define it as a constant here too... just keep it float to avoid rounding errors...

Reply:
Do you mean just the `10` ? The thing is it's then scaled by the number of CPUs, and the point of this method is to abstract it all away. For all I know this could change at some point in the kernel, so I just hide it all under a single wrapper.

What I linked is the initial value but for SMP we actually do that:
```c
max_load_balance_interval = HZ*num_online_cpus()/10;
```
Which is what the whole method does.

Reply:
Right... maybe just add the formula above (i.e. `max_load_balance_interval = HZ/10;`) in the docstring... maybe with an lxr reference to where in the kernel code this constant is defined.